### PR TITLE
Blazor updates (Part I)

### PR DESCRIPTION
**Pull request by guardrex**
Addresses #11837 

Round 1 change set:

* Re-org the TOC for Pre4 renaming.
* Move folders/topics to new locations (URLs).
* Focus on title/description (and some content) refactor for "Blazor" app model.

Suggest 🏃 *quick* 🏃 review, updates, and merge to `master` to permit the next round of updates on #11837, which I'd like to start asap.

"Razor Components" app model language is preserved to an extent here that allows us to merge this PR to `live` today (if desired for other updates on other PRs).

Content updates completing the Pre4 renaming updates are forthcoming on another PR soon and should merge with the Pre4 release.

### CODE REVIEW

**Comment by tdykstra** [comment]
Starting on it now

File: aspnetcore/blazor/index.md [hunk 1]
```diff
@@ -1,43 +1,26 @@
...
+title: Introduction to Blazor in ASP.NET Core
 author: guardrex
-description: Explore ASP.NET Core Blazor, a new way to build interactive client-side apps with .NET that run in the browser with WebAssembly.
+description: Explore ASP.NET Core Blazor, a way to build interactive client-side web UI with .NET in an ASP.NET Core app.
 monikerRange: '>= aspnetcore-3.0'
 ms.author: riande
-ms.custom: mvc
-ms.date: 04/08/2019
-uid: spa/blazor/index
+ms.custom: seoapril2019
```
Comment (tdykstra) [inline aspnetcore/blazor/index.md]: Why was this changed to seoapril2019?

File: aspnetcore/blazor/javascript-interop.md [hunk 2]
```diff
@@ -1,26 +1,26 @@
 ---
-title: Razor Components JavaScript interop
+title: Blazor JavaScript interop
```
Comment (tdykstra) [inline aspnetcore/blazor/javascript-interop.md]: For file rename
.../_static/aspnet-core-razor-components.png → .../_static/aspnet-core-razor-components.png

why leave RC in the file name?

File: aspnetcore/tutorials/build-your-first-blazor-app/samples/3.x/RazorComponents/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Build Your First Blazor App Sample (RazorComponents)
+
+This sample illustrates the basic scenarios of Blazor.
```
Comment (tdykstra) [inline aspnetcore/tutorials/build-your-first-blazor-app/samples/3.x/RazorComponents/README.md]: If this goes with a tutorial, can we add a link to the tutorial?

File: aspnetcore/blazor/debug.md [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
 ---
-# Debug Razor Components
+# Debug Blazor
 
 [Daniel Roth](https://github.com/danroth27)
 
 [!INCLUDE[](~/includes/razor-components-preview-notice.md)]
 
-Razor Components has some *very early* support for debugging client-side Blazor apps running on WebAssembly in Chrome.
+*Early* support exists for debugging client-side Blazor apps running on WebAssembly in Chrome.
```
[the remark this answers is not in the document]
Comment (guardrex) [reply aspnetcore/blazor/debug.md]: Dan's remark ... *yes* ... that's what it means.

File: aspnetcore/blazor/get-started.md [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+# Get started with Blazor
+
+By [Daniel Roth](https://github.com/danroth27) and [Luke Latham](https://github.com/guardrex)
+
+Get started with Blazor following either of the following experiences:
+
+* [Server-side Razor Components](#server-side-razor-components-experience)
+* [Client-side Blazor](#client-side-blazor-experience)
+
+## Server-side Razor Components experience
```
[the remark this answers is not in the document]
Comment (guardrex) [reply aspnetcore/blazor/get-started.md]: Because *today* (if you want to merge to `live` today) this is the case. Soon, this won't be the case, and the upcoming changes on the next PR will resolve this. This is the best we can do if we take these changes to `live` today. I'm in a bit of *limbo* on the PR *today*.

File: aspnetcore/blazor/index.md (hunk 1 above)
Comment (guardrex) [reply aspnetcore/blazor/index.md]: @wadepickett can answer.

File: aspnetcore/blazor/javascript-interop.md (hunk 2 above)
Comment (guardrex) [reply aspnetcore/blazor/javascript-interop.md]: Because *today*, this is the case. They don't see the file name, but I agree *soon* that this should be updated. The whole direction is going to go "client-side" and "server-side" shortly. *Today* tho, we're in a bit of *limbo* on these concepts. Next PR should take care of this.

File: aspnetcore/tutorials/build-your-first-blazor-app/samples/3.x/RazorComponents/README.md (hunk 3 above)
Comment (guardrex) [reply aspnetcore/tutorials/build-your-first-blazor-app/samples/3.x/RazorComponents/README.md]: Yes, I'll link it on a commit. Stand-by.

**Comment by guardrex** [comment]
@tdykstra Perhaps, the best we can do *today* if you wanted to go *live* with this round of updates. Your concerns will be fully addressed on a future PR when Pre4 releases.

The sharp edge of a razor is difficult to pass over; thus the wise say the path to 'enlightenment' is hard.* 
&nbsp;&nbsp;&nbsp;&nbsp;\- Katha Upanishad

The path to salvation between Pre3 and Pre4 is narrow and as difficult to walk as the *ASP.NET Core's Razor's edge*.
&nbsp;&nbsp;&nbsp;&nbsp;\- *WiseRex*:tm:

**Comment by tdykstra** [comment]
:shipit:

**Comment by danpadmore** [comment]
All Razor Components documentation seems to give a 404:
https://docs.microsoft.com/en-us/aspnet/core/razor-components/

Is this temporary or has it moved to a new location?

**Comment by guardrex** [comment]
@danpadmore Moved. We decided not to set up redirects for preview docs (see [discussion](https://github.com/aspnet/AspNetCore.Docs/pull/11992)). It would have cluttered up our redirects file a little bit. We think that search engines will re-index us quickly.

The coverage kicks off here now: https://docs.microsoft.com/en-us/aspnet/core/blazor/?view=aspnetcore-3.0

The TOC for Blazor should be stable going forward. The next round of updates (Part II) will only address the content and sample apps.
